### DISCUSSION TITLE
Wrangler deletes environment variables set by Terraform configuration on deployment of Pages project

### DISCUSSION THREAD

**Discussion by alex35mil** [cloudflare/workers-sdk#13479] [Feature Requests]
### Describe the solution

When managing infrastructure with Terraform and deploying Pages project with Wrangler, the latter messes up configuration set by the former.

For example, I set env vars in pages project via Terraform config and then build/deploy code using Wrangler CLI. On deployment, Wrangler deletes all env vars set by the Terraform (except for the secrets). `keep_vars/--keep-vars` is not available for Pages deployment.

**Comment by CarmenPopoviciu** [comment]
hi @alex35mil,

does your Pages project have a `wrangler.toml` configuration file? If it does that would explain why your env vars get overriden, as there can only be one config source of truth.

**Comment by alex35mil** [comment]
@CarmenPopoviciu The problem with this source of truth is that it is static and doesn't accept inputs. It doesn't allow to keep a monorepo config dry. I have to hardcode/duplicate values that are shared across the stack.
